user.php

```diff
@@ -0,0 +1,108 @@
+<?
+
+
+class User{
+
+	public $db;
+
+	private $states = [
+		'NSW',
+		'ACT',
+		'VIC',
+		'QLD',
+		'TAS',
+		'WA'
+	]
+
+	function __construct()
+	{
+		$this->db = new mysqli('127.0.0.1', 'root', 'password', 'users');
+
+	}
+
+	function check_Postcode($postcode) {
+		//.... call some external api
+		$res = external_api_call();
+		if (!$res) {
+			throw new \Exception("invalid post code");
+		}
+		return true;
+	}
+
+
+	function register($name, $password, $email, $phone = null, $postal_address = null, $postalState = null
+			, $billing_address = null, $billingState = null)
+	{
+		global $validator;
+
+		//check for an existing account
+		$q = "SELECT * FROM user WHERE email = '$email'";
```

Comment:
This is potentially vulnerable to sql injection. A prepared statement should be used instead.

Reply:
This logic should be in a separate method, potentially in the validator.

Reply:
And if the intention is just to check if the email exists in the database, there is no need to use *

``` SELECT id FROM user WHERE email = '$email'";```  should be enough